src/commands/paranet/start-paranet-sync-commands.js

```diff
@@ -0,0 +1,112 @@
+import Command from '../command.js';
+import { ERROR_TYPE, PARANET_SYNC_FREQUENCY_MILLS } from '../../constants/constants.js';
+
+class StartParanetSyncCommands extends Command {
+    constructor(ctx) {
+        super(ctx);
+        this.commandExecutor = ctx.commandExecutor;
+        this.ualService = ctx.ualService;
+        this.blockchainModuleManager = ctx.blockchainModuleManager;
+        this.repositoryModuleManager = ctx.repositoryModuleManager;
+
+        this.errorType = ERROR_TYPE.PARANET.START_PARANET_SYNC_ERROR;
+    }
+
+    async execute() {
+        const operationId = this.operationIdService.generateId();
+
+        this.logger.info(
+            `Paranet sync: Starting Paranet sync command for operation id: ${operationId}`,
+        );
+
+        await this.commandExecutor.delete('paranetSyncCommand');
+
+        const promises = [];
+        this.config.assetSync?.syncParanets.forEach(async (paranetId) => {
+            if (!this.ualService.isUal(paranetId)) {
+                this.logger.info(`Paranet sync: paranetId ${paranetId} is not a valid UAL!`);
+                return Command.empty();
+            }
+
+            const contractKaCount = await this.blockchainModuleManager.getKnowledgeAssetsCount(
+                paranetId,
+            );
+            const cachedKaCount =
+                (await this.repositoryModuleManager.getParanetById(paranetId)?.kaCount) ?? 0;
+
+            if (cachedKaCount === contractKaCount) {
+                this.logger.info(
+                    `Paranet sync: KA count from contract and in DB is the same, nothing to sync!`,
+                );
+                return Command.empty();
+            }
+
+            this.logger.info(
+                `Paranet sync: Syncing ${contractKaCount - cachedKaCount + 1} assets...`,
+            );
+
+            const kaToUpdate = [];
+            for (let i = cachedKaCount; i <= contractKaCount; i += 50) {
+                const nextKaArray = this.blockchainModuleManager.getKnowledgeAssetsWithPagination(
+                    paranetId,
+                    i,
+                    50,
+                );
+                if (!nextKaArray.length) break;
+                kaToUpdate.push(...nextKaArray);
+            }
+
+            kaToUpdate
+                .map((ka) => ka.tokenId)
+                .forEach((tokenId) => {
+                    const commandData = {
+                        paranetId,
+                        operationId,
+                        tokenId,
+                    };
+
+                    promises.append(
+                        this.commandExecutor.add({
```

Comment:
I think that it would be better to change the logic here. This command will potentially schedule a lot of paranet-sync-commands (if we have a lot of assets in Paranet and node started syncing it for the first time.). Previously we had a problem when scheduling a lot of commands in command executor - operational db become too slow... 
I recommend to change the logic so that start-paranet-sync-command is scheduled as permanent command and executed just once during node start. It's purpose is to schedule new paranet-sync-command for each Paranet user defined in configuration. 
paranet-sync-command should have logic to get and store all assets connected to that paranet. 
We did logic like this in epoch-check-command/blockchain-epoch-check-command and get-latest-service-agreement-data/blockchain-get-latest-service-agreement-data please check it out.